tools/system_libs.py

```diff
@@ -650,7 +686,7 @@ def get_default_variation(cls, **kwargs):
     return super().get_default_variation(is_asan=settings.USE_ASAN, **kwargs)
 
 
-class libcompiler_rt(MTLibrary):
+class libcompiler_rt(MTLibrary, SjLjLibrary):  # ASan uses setjmp/longjmp
```

Comment:
Oh didn't know that. Then we should make asan `SjLjLibrary` instead I guess?

Reply:
Oh no, this is necessary for another reason, because of this part: https://github.com/emscripten-core/emscripten/pull/14976/files#diff-444eb2dc4484af33db2401becfb685bbab1916088c79d969d7c87406d9a39e27R71-R90

This `__USING_WASM_SJLJ__` is enabled when `is_wasm` in `SjLjLibrary` is set. We can't compile this part unconditionally because [this line](https://github.com/emscripten-core/emscripten/pull/14976/files#diff-444eb2dc4484af33db2401becfb685bbab1916088c79d969d7c87406d9a39e27R87) containing a call to `__builtin_wasm_throw` will not compile unless `is_wasm` in `SjLjLibrary` is set, which sets `SUPPORT_LONGJMP=wasm`, which in turn [sets](https://github.com/emscripten-core/emscripten/pull/14976/files#diff-e6615de4338b98e08471c53f2d6fc250dc6ca22cd8c303ac33ab259c18ae68d2R252-R253) `-mllvm -wasm-enable-sjlj`, which enables EH capability in LLVM.